File: be/src/sub-tasks/entities/sub-task.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column as ColumnField,
  ManyToOne,
  JoinColumn,
  Index,
} from 'typeorm';
import { Task } from 'src/tasks/entities/task.entity';

@Entity('subtasks')
@Index(['taskId'])
export class SubTask {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ColumnField({ name: 'title' })
  title: string;

  @ColumnField({ default: false, name: 'completed' })
  completed: boolean;

  @ColumnField({ type: 'timestamp', nullable: true, name: 'completed_at' })
  completedAt?: Date;

  @ColumnField({ type: 'decimal', precision: 10, scale: 3, default: 0, name: 'position' })
  position: string;

  @ColumnField({ name: 'task_id', type: 'uuid' })
  taskId: string;

  @ManyToOne(() => Task, (task) => task.subtasks, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'task_id' })
  task: Task;
}

File: backend-java/src/main/java/com/qlda/backendjava/subtasks/entity/SubTaskEntity.java
package com.qlda.backendjava.subtasks.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.qlda.backendjava.tasks.entity.TaskEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "subtasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class SubTaskEntity {

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @Column(columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "completed", nullable = false)
    private Boolean completed = false;

    @Column(name = "completed_at", nullable = true)
    private LocalDateTime completedAt;

    @Column(name = "position", precision = 10, scale = 3, nullable = false)
    private BigDecimal position = BigDecimal.ZERO;

    @Column(name = "task_id", nullable = false)
    private String taskId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task_id", insertable = false, updatable = false)
    @JsonIgnore
    private TaskEntity task;
}

